pkg/planner/core/casetest/physicalplantest/testdata/plan_suite_out.json

```diff
@@ -2317,43 +2317,155 @@
+          "  └─IndexReader(Probe) 9990.00 root  index:IndexFullScan",
+          "    └─IndexFullScan 9990.00 cop[tikv] table:t1, index:idx_a(a) keep order:false, stats:pseudo"
+        ]
       },
       {
         "SQL": "SELECT a FROM (SELECT a, (SELECT t2.b FROM t t2, t t3 WHERE t2.a = t3.a AND t1.a = t2.a LIMIT 1) t FROM t t1) t",
-        "Best": "IndexReader(Index(t.f)[[NULL,+inf]])"
+        "Best": "IndexReader(Index(t.f)[[NULL,+inf]])",
+        "Plan": [
+          "IndexReader 10000.00 root  index:IndexFullScan",
+          "└─IndexFullScan 10000.00 cop[tikv] table:t1, index:idx_a(a) keep order:false, stats:pseudo"
+        ]
       },
       {
         "SQL": "SELECT a FROM t WHERE b IN (SELECT t2.b FROM t t2, t t3 WHERE t2.a = t3.a)",
-        "Best": "LeftHashJoin{TableReader(Table(t))->MergeInnerJoin{TableReader(Table(t))->TableReader(Table(t))}(test.t.a,test.t.a)->HashAgg}(test.t.b,test.t.b)"
+        "Best": "LeftHashJoin{TableReader(Table(t))->MergeInnerJoin{TableReader(Table(t))->TableReader(Table(t))}(test.t.a,test.t.a)->HashAgg}(test.t.b,test.t.b)",
+        "Plan": [
+          "HashJoin 9980.01 root  inner join, equal:[eq(test.t.b, test.t.b)]",
+          "├─HashAgg(Build) 7984.01 root  group by:test.t.b, funcs:firstrow(test.t.b)->test.t.b",
+          "│ └─HashJoin 12475.01 root  inner join, equal:[eq(test.t.a, test.t.a)]",
+          "│   ├─IndexReader(Build) 9990.00 root  index:IndexFullScan",
+          "│   │ └─IndexFullScan 9990.00 cop[tikv] table:t3, index:idx_a(a) keep order:false, stats:pseudo",
+          "│   └─TableReader(Probe) 9980.01 root  data:Selection",
+          "│     └─Selection 9980.01 cop[tikv]  not(isnull(test.t.a)), not(isnull(test.t.b))",
+          "│       └─TableFullScan 10000.00 cop[tikv] table:t2 keep order:false, stats:pseudo",
+          "└─TableReader(Probe) 9990.00 root  data:Selection",
+          "  └─Selection 9990.00 cop[tikv]  not(isnull(test.t.b))",
+          "    └─TableFullScan 10000.00 cop[tikv] table:t keep order:false, stats:pseudo"
+        ]
       },
       {
         "SQL": "SELECT a FROM t WHERE EXISTS (SELECT t2.b FROM t t2, t t3 WHERE t2.a = t3.a AND t2.b=t.b)",
-        "Best": "LeftHashJoin{TableReader(Table(t))->MergeInnerJoin{TableReader(Table(t))->TableReader(Table(t))}(test.t.a,test.t.a)}(test.t.b,test.t.b)"
+        "Best": "LeftHashJoin{TableReader(Table(t))->MergeInnerJoin{TableReader(Table(t))->TableReader(Table(t))}(test.t.a,test.t.a)}(test.t.b,test.t.b)",
+        "Plan": [
+          "HashJoin 7992.00 root  semi join, left side:TableReader, equal:[eq(test.t.b, test.t.b)]",
+          "├─TableReader(Build) 9990.00 root  data:Selection",
+          "│ └─Selection 9990.00 cop[tikv]  not(isnull(test.t.b))",
+          "│   └─TableFullScan 10000.00 cop[tikv] table:t keep order:false, stats:pseudo",
+          "└─HashJoin(Probe) 12475.01 root  inner join, equal:[eq(test.t.a, test.t.a)]",
```

Comment:
the merge join becomes hash join, is it reasonable?

Reply:
It just adds the explain result for this test. It is more readable. But it is the same as the master's result.